opentracing-spring-cloud/src/test/java/io/opentracing/contrib/spring/cloud/log/LoggingAutoConfigurationTest.java

```diff
@@ -0,0 +1,161 @@
+    private static final Log commonsLog = LogFactory.getLog(Controller.class);
+    private static final Logger slf4jLog = LoggerFactory.getLogger(Controller.class);
+    private static final java.util.logging.Logger julLog = java.util.logging.Logger.getLogger(Controller.class.getName());
+    private static final org.apache.log4j.Logger log4j = org.apache.log4j.Logger.getLogger(Controller.class);
+
+    private ContextData contextData() {
+      ContextData contextData = new ContextData();
+      contextData.setTimestamp(System.currentTimeMillis() * 1000);
+      contextData.setThread(Thread.currentThread().getName());
+      return contextData;
+    }
+
+    @RequestMapping("/jcl")
+    public ContextData jcl() {
+      commonsLog.info(LOG);
+      return contextData();
+    }
+    @RequestMapping("/slf4j")
+    public ContextData slf4j() {
+      slf4jLog.info(LOG);
+      return contextData();
+    }
+    @RequestMapping("/jul")
+    public ContextData jul() {
+      julLog.info(LOG);
+      return contextData();
+    }
+    @RequestMapping("/log4j")
+    public ContextData log4j() {
+      log4j.info(LOG);
+      return contextData();
+    }
+  }
+
+  @Autowired
+  private TestRestTemplate restTemplate;
+  @Autowired
+  private MockTracer mockTracer;
+
+  @After
+  public void after() {
+    mockTracer.reset();
+  }
+
+  @Test
+  public void testJcl() {
+    ResponseEntity<ContextData> response = restTemplate.getForEntity("/jcl", ContextData.class);
+    assertLogging(response.getBody());
+  }
+
+  @Test
+  public void testSlf4j() {
+    ResponseEntity<ContextData> response = restTemplate.getForEntity("/slf4j", ContextData.class);
+    assertLogging(response.getBody());
+  }
+
+  @Test
+  public void testJul() {
+    ResponseEntity<ContextData> response = restTemplate.getForEntity("/jul", ContextData.class);
+    assertLogging(response.getBody());
+  }
+
+  @Test
+  public void testLog4j() {
+    ResponseEntity<ContextData> response = restTemplate.getForEntity("/log4j", ContextData.class);
+    assertLogging(response.getBody());
+  }
+
+  private void assertLogging(ContextData contextData) {
+    List<MockSpan> mockSpans = mockTracer.finishedSpans();
+    assertEquals(1, mockSpans.size());
+    MockSpan mockSpan = mockSpans.get(0);
+    // preHandle log
+    // standard log
+    // afterCompletion log
+    assertEquals(3, mockSpan.logEntries().size());
+    LogEntry logEntry = mockSpan.logEntries().get(1);
+    assertEquals(4, logEntry.fields().size());
+    assertEquals(Controller.class.getName(), logEntry.fields().get("logger"));
+    assertEquals(LOG, logEntry.fields().get("message"));
+    assertEquals(contextData.getThread(), logEntry.fields().get("thread"));
+    assertEquals(Level.INFO.toString(), logEntry.fields().get("level"));
+    // now <= timestamp < now + 10ms
+    assertTrue(logEntry.timestampMicros() >= contextData.getTimestamp() && logEntry.timestampMicros() < contextData.getTimestamp() + 10 * 1000);
```

Comment:
Shouldn't the check be the other way around? The log entry's timestamp will either be less or equal to the context data's timestamp?

Reply:
good catch, before I was setting the timestamp before logging. CI failed on this